libfreerdp/crypto/crypto.c

```diff
@@ -289,6 +289,30 @@ BYTE* crypto_cert_hash(X509* xcert, const char* hash, UINT32* length)
 	return fp;
 }
 
+char* cryto_cert_hash_use_signature_alg(X509* xcert)
```

Comment:
@akarl10 ok, if you want to use it this way better use a `char* crypto_cert_get_signature_alg(X509* xcert, char* buffer, size_t buffersize)` function signature

Reply:
another option would be to return a `WINPR_MD_TYPE`

Reply:
I think maybe the second option would be better. The buffer could be hidden inside the function and at the end of the day WINPR_MD_TYPE is what we need. `winpr_md_type_to_string` returns a pointer to a const char*, so we don't need to bother about the allocation

Reply:
good point

Reply:
is it allowed to include something from winpr in inlcude/freerdp/crypto? I get `crypto.h:60:21: error: unknown type name ‘WINPR_MD_TYPE’`

Reply:
@akarl10 yes, just not the other way around